### ISSUE TITLE
Error on mixed format 

### ISSUE THREAD

**Issue by laBecasse** [przemyslawpluta/node-youtube-dl#221]
There is a error when we use mixed format like:
```
    let video = youtubedl(url, ['--format=137+140'], {cwd: __dirname, maxBuffer: Infinity})
```

the error is:

```
url.js:150
    throw new ERR_INVALID_ARG_TYPE('url', 'string', url);
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received type undefined
    at Url.parse (url.js:150:11)
    at Object.urlParse [as parse] (url.js:144:13)
    at processData (/node_modules/youtube-dl/lib/youtube-dl.js:47:21)
    at getInfo (/node_modules/youtube-dl/lib/youtube-dl.js:109:9)
    at done (/node_modules/youtube-dl/lib/youtube-dl.js:295:9)
    at done (/node_modules/youtube-dl/lib/youtube-dl.js:202:9)
    at ChildProcess.exithandler (child_process.js:280:7)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
```

**Comment by saltomodules** [comment]
Had the same issue .

This is caused by using a node stream to download the video which this lib does by default. If you give up a mixed format, two separate downloads need to be done and ffmpeg needs to mux them together afterwards.

This is impossible to do with one single node stream, which does just one thing: download one single video file. 

We worked around this issue by calling youtube-dl directly with youtubedl.exec(url). In that way all downloading and muxing is handled inside youtube-dl itself. You lose the perks of a node stream like progress events etc, but you can download mixed formats.

**Comment by Kikobeats** [comment]
Thanks for the explanation @saltomodules 🙂

**Comment by kyleratti** [comment]
@Kikobeats: Any problem with adding a note about this to `README.md`? My original project last year was plagued by this issue and I just spent several hours debugging a very similar issue on my new project. It's pure luck I stumbled across this post.

@saltomodules: Many thanks for taking the time to look at the issue and post your detailed response. People like you are what make programming great even when troubleshooting gets frustrating.
